### PR TITLE
Fix array proxy object convertion

### PR DESCRIPTION
**Pull request by PatriceJiang**
Re: https://github.com/cocos/3d-tasks/issues/15256

### Changelog

* `se::Object` add interface for proxy:  `createProxyTarget`
* Simplify proxy for Array wrapper returned from native
* Pass a pure js object to `pipeline*.initialize` functions

-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

File: cocos/rendering/index.jsb.ts [hunk 1]
```diff
@@ -54,6 +54,24 @@ export const ReflectionProbeFlow = nr.ReflectionProbeFlow;
...
 export { PipelineEventType } from './pipeline-event';
 
+interface IRenderFlowInfo {
+    name: string;
+    priority: number;
+    stages: any[];
+    tag: number;
+}
+
+interface IRenderPipelineInfo {
```
Comment (dumganhar) [inline cocos/rendering/index.jsb.ts:64]: Do we need to export these interfaces?

Comment (PatriceJiang) [reply cocos/rendering/index.jsb.ts:64]: It's not exported in `rendering/index.ts`

File: cocos/asset/assets/asset.jsb.ts [hunk 2]
```diff
@@ -48,21 +48,22 @@ assetProto._ctor = function () {
 
     this._callbackTable = js.createMap(true);
     this._file = null;
+    this._uuid = ''; // @deprecated since v3.5.0
```
Comment (dumganhar) [inline cocos/asset/assets/asset.jsb.ts]: `_uuid` is a jsb property. no need to initialize it here.
```c++
bool js_register_cc_Asset(se::Object* obj) {
    auto* cls = se::Class::create("Asset", obj, __jsb_cc_CCObject_proto, _SE(js_new_cc_Asset)); 
    
    cls->defineStaticProperty("__isJSB", se::Value(true), se::PropertyAttribute::READ_ONLY | se::PropertyAttribute::DONT_ENUM | se::PropertyAttribute::DONT_DELETE);
    cls->defineProperty("_native", _SE(js_cc_Asset__native_get), _SE(js_cc_Asset__native_set)); 
    cls->defineProperty("_nativeUrl", _SE(js_cc_Asset__nativeUrl_get), _SE(js_cc_Asset__nativeUrl_set)); 
    cls->defineProperty("_uuid", _SE(js_cc_Asset__uuid_get), _SE(js_cc_Asset__uuid_set)); 
//...
```

File: native/cocos/bindings/jswrapper/v8/Object.h [hunk 3]
```diff
@@ -177,6 +177,14 @@ class Object final : public RefCounter {
      */
     static Object *createObjectWithConstructor(se::Object *constructor, const ValueArray &args);
 
+    /**
+     * @brief Get the Proxy Target object
```
Comment (dumganhar) [inline native/cocos/bindings/jswrapper/v8/Object.h]: Get -> Gets

File: native/cocos/bindings/jswrapper/v8/Object.h [hunk 4]
```diff
@@ -177,6 +177,14 @@ class Object final : public RefCounter {
...
+     *
+     * @param proxy
+     * @return Object*
```
Comment (dumganhar) [inline native/cocos/bindings/jswrapper/v8/Object.h]: Comments are empty.

File: cocos/asset/assets/asset.jsb.ts (hunk 2 above)
Comment (PatriceJiang) [reply cocos/asset/assets/asset.jsb.ts]: fixed

File: native/cocos/bindings/jswrapper/v8/Object.h (hunk 4 above)
Comment (PatriceJiang) [reply native/cocos/bindings/jswrapper/v8/Object.h]: fixed

File: native/cocos/bindings/jswrapper/v8/Object.h (hunk 3 above)
Comment (PatriceJiang) [reply native/cocos/bindings/jswrapper/v8/Object.h]: fixed

**Review by dumganhar — APPROVED** [review]

**Comment by dumganhar** [comment]
@cocos-robot run test cases

**Comment by dumganhar** [comment]
@cocos-robot run test cases

**Pull request merged**